aws-opentelemetry-distro/src/amazon/opentelemetry/distro/aws_opentelemetry_configurator.py

```diff
@@ -634,10 +634,11 @@ def _validate_and_fetch_logs_header() -> OtlpLogHeaderSetting:
     logs_headers = os.environ.get(OTEL_EXPORTER_OTLP_LOGS_HEADERS)
 
     if not logs_headers:
-        _logger.warning(
-            "Improper configuration: Please configure the environment variable OTEL_EXPORTER_OTLP_LOGS_HEADERS "
-            "to include x-aws-log-group and x-aws-log-stream"
-        )
+        if not _is_lambda_environment():
+            _logger.warning(
+                "Improper configuration: Please configure the environment variable OTEL_EXPORTER_OTLP_LOGS_HEADERS "
+                "to include x-aws-log-group and x-aws-log-stream"
+            )
         return OtlpLogHeaderSetting(None, None, None, False)
```

Comment:
Here we are saying using this header as a valid header for Lambda, but still return `is_valid=false`. I think we should get rid of `is_valid` in `OtlpLogHeaderSetting` because it is redundant since we also return values for logGroup/logStream. Wherever we use the `is_valid` value, just replace it to check that logGroup and logStream are defined.